http/info.go

```diff
@@ -86,10 +91,22 @@ func TraceInfoFromHeader(header HeaderGetter) (rv TraceInfo) {
 			return rv
 		}
 
+		bm := map[string]string{}
+		if baggage != "" {
+			for _, kv := range strings.Split(baggage, ",") {
+				parts := strings.Split(kv, "=")
+				for _, b := range allowedBaggage {
+					if parts[0] == b {
+						bm[parts[0]] = parts[1]
```

Comment:
Break out of the loop as well. alternatively, make a small func `stringsContains`